web/src/containers/Header/index.tsx

```diff
@@ -68,24 +68,28 @@ class Head extends React.Component<Props> {
         return (
             <header className={`pg-header`}>
                 <div className={`pg-container pg-header__content ${tradingCls}`}>
-                    <div
-                        className={`pg-sidebar__toggler ${mobileWallet && 'pg-sidebar__toggler-mobile'}`}
-                        onClick={this.openSidebar}>
-                        <span className="pg-sidebar__toggler-item" />
-                        <span className="pg-sidebar__toggler-item" />
-                        <span className="pg-sidebar__toggler-item" />
-                    </div>
-                    <div onClick={(e) => this.redirectToLanding()} className="pg-header__logo">
-                        <Logo />
-                    </div>
+                    {!useSharedLayout() &&
+                        <>
+                            <div
+                                className={`pg-sidebar__toggler ${mobileWallet && 'pg-sidebar__toggler-mobile'}`}
+                                onClick={this.openSidebar}>
+                                <span className="pg-sidebar__toggler-item" />
+                                <span className="pg-sidebar__toggler-item" />
+                                <span className="pg-sidebar__toggler-item" />
+                            </div>
+                            <div onClick={(e) => this.redirectToLanding()} className="pg-header__logo">
+                                <Logo />
+                            </div>
+                        </>
+                    }
```

Comment:
maybe let's rewrite this component using functional approach and memorize this block using useMemo